ev3dev/evb.py

```diff
@@ -0,0 +1,128 @@
+
+"""
+An assortment of classes modeling specific features of the EVB brick.
+"""
+
+from .core import *
+
+
+OUTPUT_A = 'outA'
+OUTPUT_B = 'outB'
+OUTPUT_C = 'outC'
+OUTPUT_D = 'outD'
+
+INPUT_1 = 'in1'
+INPUT_2 = 'in2'
+INPUT_3 = 'in3'
+INPUT_4 = 'in4'
+
+
+class Button(ButtonEVIO):
+    """
+    EVB Buttons
+    """
+
+    _buttons = {
+            'up': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 103},
+            'down': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 108},
+            'left': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 105},
+            'right': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 106},
+            'enter': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 28},
+            'backspace': {'name': '/dev/input/by-path/platform-evb-buttons-event', 'value': 14},
+        }
+
+    @property
+    @staticmethod
+    def on_up(state):
+        """
+        This handler is called by `process()` whenever state of 'up' button
+        has changed since last `process()` call. `state` parameter is the new
+        state of the button.
+        """
+        pass
+
+    @staticmethod
+    def on_down(state):
+        """
+        This handler is called by `process()` whenever state of 'down' button
+        has changed since last `process()` call. `state` parameter is the new
+        state of the button.
+        """
+        pass
+
+    @staticmethod
+    def on_left(state):
+        """
+        This handler is called by `process()` whenever state of 'left' button
+        has changed since last `process()` call. `state` parameter is the new
+        state of the button.
+        """
+        pass
+
+    @staticmethod
+    def on_right(state):
+        """
+        This handler is called by `process()` whenever state of 'right' button
+        has changed since last `process()` call. `state` parameter is the new
+        state of the button.
+        """
+        pass
+
+    @staticmethod
+    def on_enter(state):
+        """
+        This handler is called by `process()` whenever state of 'enter' button
```

Comment:
What is this for?

Reply:
This is so you can do things like
```
def foobar():
    print("The user pressed ENTER")

btn = Button()
btn.on_enter = foobar

while True:
    btn.process()
```

I'm working on making this event driven as part of the buttons API work...today you have to call it in a while loop like that to make it do what you want.

Reply:
Oh, I see. Damn, Python is weird :laughing: Is that a common idiom? Is there not an event framework of some sort?

Either way, I don't think it's blocking.

Reply:
This project is the only place I've seen this approach used...but I've only been writing python for 4 years so who knows maybe it is wildly popular and I just haven't been exposed to it.